src/view/upcastwriter.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module module:engine/view/upcastwriter
+ */
+
+import Element from './element';
+
+/**
+ * View upcast writer class. Provides set of methods used to properly manipulate nodes attached to
+ * {@link module:engine/view/view~View view instance}. It should be only used to manipulate non-semantic view
+ * (view created from HTML string). For view which was downcasted from the {@link module:engine/model/model~Model model}
+ * see {@link module:engine/view/downcastwriter~DowncastWriter downcast writer}.
+ */
+export default class UpcastWriter {
+	/**
+	 * Clones provided element.
+	 *
+	 * @see module:engine/view/element~Element#_clone
+	 * @param {module:engine/view/element~Element} element Element to be cloned.
+	 * @param {Boolean} [deep=false] If set to `true` clones element and all its children recursively. When set to `false`,
+	 * element will be cloned without any children.
+	 * @returns {module:engine/view/element~Element} Clone of this element.
+	 */
+	clone( element, deep = false ) {
+		return element._clone( deep );
+	}
+
+	/**
+	 * Appends a child node or a list of child nodes at the end of this node
+	 * and sets the parent of these nodes to this element.
+	 *
+	 * @see module:engine/view/element~Element#_appendChild
+	 * @param {module:engine/view/element~Element|module:engine/view/documentfragment~DocumentFragment} element Element
+	 * to which items will be appended.
+	 * @param {module:engine/view/item~Item|Iterable.<module:engine/view/item~Item>} items Items to be inserted.
+	 * @fires module:engine/view/node~Node#change
```

Comment:
We (at least I ;)) usually write `event:change`. Does the line above create proper docs?

BTW., those change events on view tree are so internal that I am not sure if it is worth documenting. @pjasiun ?

Reply:
From what I see, both produce the same link `.../module_engine_view_node-Node.html#event:change` but I will update it to make it consistent.

However, I just discovered it doesn't work properly in our docs 😱 The generated links are e.g. [`...api/module_engine_conversion_upcastdispatcher-UpcastDispatcher.html#event:element`](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_conversion_upcastdispatcher-UpcastDispatcher.html#event:element), while to working one is [`...api/module_engine_conversion_upcastdispatcher-UpcastDispatcher.html#event-element`](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_conversion_upcastdispatcher-UpcastDispatcher.html#event-element) (so `-` instead of `:` in event name). I will report it.